Framework/Algorithms/src/ReflectometryWorkflowBase2.cpp

```diff
@@ -685,5 +680,70 @@ ReflectometryWorkflowBase2::getRunNumber(MatrixWorkspace const &ws) const {
   return "";
 }
 
+std::string
+ReflectometryWorkflowBase2::convertProcessingInstructionsToWorkspaceIndexes(
+    const std::string &instructions, MatrixWorkspace_const_sptr ws) const {
+  std::string converted = "";
+  std::string currentNumber = "";
+  std::string ignoreThese = "-,:+";
+  for (auto i = 0u; i < instructions.size(); ++i) {
+    if (std::find(ignoreThese.begin(), ignoreThese.end(), instructions[i]) !=
+        ignoreThese.end()) {
+      // Found a spacer so add currentNumber to converted after seperator
+      converted.append(workspaceIndexesToSpecNum(currentNumber, ws));
+      converted.push_back(instructions[i]);
+      currentNumber = "";
+    } else {
+      currentNumber.push_back(instructions[i]);
+    }
+  }
+  // Add currentNumber onto converted
+  converted.append(workspaceIndexesToSpecNum(currentNumber, ws));
+  return converted;
+}
+
+std::string ReflectometryWorkflowBase2::workspaceIndexesToSpecNum(
+    const std::string &num, MatrixWorkspace_const_sptr ws) const {
+  if (num == "0") {
+    return "0";
+  }
+  auto specNum = atoi(num.c_str());
+  auto x = ws->x(0);
+  std::string wsIdx = std::to_string(
+      ws->getIndexFromSpectrumNumber(static_cast<specnum_t>(specNum)));
+  return wsIdx;
+}
+
+std::string ReflectometryWorkflowBase2::convertWorkspaceIndexProcInstToSpecNum(
+    const std::string &instructions,
+    Mantid::API::MatrixWorkspace_const_sptr ws) const {
+  std::string converted = "";
+  std::string currentNumber = "";
+  std::string ignoreThese = "-,:+";
+  for (auto i = 0u; i < instructions.size(); ++i) {
+    if (std::find(ignoreThese.begin(), ignoreThese.end(), instructions[i]) !=
+        ignoreThese.end()) {
+      // Found a spacer so add currentNumber to converted after seperator
+      converted.append(specNumToWorkspaceIndexes(currentNumber, ws));
+      converted.push_back(instructions[i]);
+      currentNumber = "";
+    } else {
+      currentNumber.push_back(instructions[i]);
+    }
+  }
+  // Add currentNumber onto converted
+  converted.append(specNumToWorkspaceIndexes(currentNumber, ws));
+  return converted;
+}
+std::string ReflectometryWorkflowBase2::specNumToWorkspaceIndexes(
+    const std::string &num, Mantid::API::MatrixWorkspace_const_sptr ws) const {
+  if (num == "0") {
```

Comment:
Why do we return `0` if the spectrum number is `0`? This is an invalid input so would it be better to throw? Are there other invalid inputs we need to handle?